src/Controls/src/Core/Application/Application.cs

```diff
@@ -101,6 +101,15 @@ public IAppLinks AppLinks
 
 				_singleWindowMainPage = value;
 
+				if (value is not null)
```

Comment:
this fixes another bug. resources changed at application level weren't propagated to MainPage

Reply:
I think if you are just using MainPage, then you get parented to a window during the creating of the UI - so once the app is ready, it calls CreateWindow and assigns the MainPage to the window.

The `_singleWindowMainPage` is just a placeholder while the app is still being constructed. Changing MainPage later literally just replaces the content of the first window:

```cs
Windows[0].Page = value;
```

What you have observed might just be because MainPage really is a delayed construction and assigned. Now this is to say it may be fine what you are doing here, but will it result in a double listener registration once the window is created and the MainPage parented to the window?

Reply:
maybe, but that doesn't work in unit tests